pkg/cloudprovider/aws/suite_test.go

```diff
@@ -389,6 +389,21 @@ var _ = Describe("Allocation", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(constraints.SecurityGroupSelector).To(Equal(map[string]string{"kubernetes.io/cluster/test-cluster": "*"}))
 		})
+		It("should use DefaultInstanceProfile if none provided in Provider", func() {
+			provider = &v1alpha1.AWS{}
+			provisioner = ProvisionerWithProvider(&v1alpha5.Provisioner{ObjectMeta: metav1.ObjectMeta{Name: v1alpha5.DefaultProvisioner.Name}}, provider)
+			provisioner.SetDefaults(ctx)
+
+			constraints, err := v1alpha1.Deserialize(&provisioner.Spec.Constraints)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(constraints.InstanceProfile).To(Equal("default-profile"))
+		})
+		It("should use overridden instanceProfile if provided in Provider", func() {
+			provisioner.SetDefaults(ctx)
+			constraints, err := v1alpha1.Deserialize(&provisioner.Spec.Constraints)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(constraints.InstanceProfile).To(Equal("test-instance-profile"))
```

Comment:
Curious how this test passes. Don't you need to set this in the provisioner before defaulting?

Reply:
That's because this test is using the Provisioner set up earlier, which was specifying the InstanceProfile. I'll probably flip this test if we flip the logic and make the default required.